pkg/scheduler/scheduler.go

```diff
@@ -329,6 +342,77 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 	return wait.KeepGoing
 }
 
+func (s *Scheduler) addFairSharingEntryPenalty(ctx context.Context, e *entry) error {
+	var lq kueue.LocalQueue
+	lqKey := client.ObjectKey{Namespace: e.Obj.Namespace, Name: string(e.Obj.Spec.QueueName)}
+	if err := s.client.Get(ctx, lqKey, &lq); err != nil {
+		return err
+	}
+
+	afsStatus, err := getFairSharingStatus(&lq)
+	if err != nil {
+		return err
+	}
+
+	penalty, err := calcFairSharingEntryPenalty(s.admissionFairSharing, e)
+	if err != nil {
+		return err
+	}
+
+	applyPenaltyToStatus(afsStatus, penalty)
+	afsStatus.LastUpdate = metav1.Now()
+	lq.Status.FairSharing.AdmissionFairSharingStatus = afsStatus
+	if err := s.client.Status().Update(ctx, &lq); err != nil {
```

Comment:
Does it actually indicate a dedicated status update for adding penalty? IIUC this could still result in some race conditions? It would be great to add an integration test to show the penalty is respected without any delay.